TAGradingServer/account/admin-gradeable.php

```diff
@@ -881,6 +881,26 @@ function deleteForm() {
             return false;
         }
     }
+
+    function createCrossBrowserJSDate(val){
+        // Create a Date object that is cross-platform supported.
+        // Safari's Date object constructor is more restrictive that Chrome and 
+        // Firefox and will treat some dates as invalid.  Implementation details
+        // vary by browser and JavaScript engine.
+        // To solve this, we use Moment.js to standardize the parsing of the 
+        // datetime string and convert it into a RFC2822 / IETF date format.
+        //
+        // For example, ""2013-05-12 20:00:00"" is converted with Moment into 
+        // "Sun May 12 2013 00:00:00 GMT-0500 (EST)" and correctly parsed by
+        // Safari, Chrome, and Firefox.
+        //
+        // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse
+        // Ref: http://stackoverflow.com/questions/16616950/date-function-returning-invalid-date-in-safari-and-firefox
+        var timeParseString = "YYYY-MM-DD HH:mm:ss.S" // Expected string format given by the server
+        var momentDate = moment(val, timeParseString) // Parse raw datetime string
+        return new Date(momentDate.toString()) // Convert moment into RFC2822 and construct browser-specific jQuery Date object
```

Comment:
I think I'd rather just see `val = val.replace(/ /g, "T") + "Z"` than including a 4000 line library as in minimal testing this seems to be accepted in Safari, Chrome, and Firefox and follows the ISO format so it won't be browser specific (unless the browser is absolute garbage and doesn't follow standards).